_source/_posts/2022-06-17-react-firebase-serverless.md

```diff
@@ -0,0 +1,713 @@
+
+Finally, click the **Build**>**Authentication** in the side nav and click **Get Started**. Pressing the **Get Started** button generates a **Web API Key** that you will need to use the auth features. 
+
+## Create your front end in React
+
+Next, create your React app.  
+
+```console
+npx create-react-app@5 okta-firebase
+cd okta-firebase
+```
+
+The npx command will scaffold a new React version 18 application for you.  
+
+Next, you will need to install a few dependencies to help you.
+
+```console
+npm i @okta/okta-react@6.4.3
+npm i @okta/okta-auth-js@6.2.0
+npm i react-router-dom@5.3.0
+npm i bootstrap@5.1.3
+npm i firebase@9.8.1
+```
+
+First, you want to install the `@okta/okta-react` package to assist in authenticating users and obtaining the tokens from the Okta authorization server. This package will help you access the authentication state, direct users to the Okta login page, and handle any callbacks.
+
+`@okta/okta-react` relies on the `@okta/okta-auth-js` package, so you need to install it.
+
+Next, you want to install `react-router-dom`. This package will set up the callback route for you and any other routes you may need.  
+
+Finally, you will use the `firebase` package to call the various platform features in Firebase, such as functions and authentication.  
+
+Add a file called `.env` in your root directory and replace the code with the following.
+
+```
+REACT_APP_OKTA_ISSUER=https://{yourOktaDomain}/oauth2/default
+REACT_APP_OKTA_CLIENTID={yourClientID}
+REACT_APP_FIREBASE_APIKEY={yourFirebaseAPIKey}
+REACT_APP_FIREBASE_AUTHDOMAIN={yourFirebaseAuthDomain}
+REACT_APP_FIREBASE_PROJECTID={yourFirebaseProjectID}
+REACT_APP_FIREBASE_STORAGEBUCKET={yourFirebaseStorageBucket}
+REACT_APP_FIREBASE_MESSAGINGSENDERID={yourFirebaseMessagingSenderID}
+REACT_APP_FIREBASE_APPID={yourFirebaseAppID}
+REACT_APP_ENV=production
+```
+
+You obtained the Okta values when you created your application using the Okta CLI earlier. Your Okta domain is part of the Issuer. The Firebase values came from the configuration you copied when you first created your application.
+
+There is currently a known error in React 18 with the Okta React library where multiple rerenders can lead to an error message in the `oktaAuth` object. While fixing the error is ongoing, you can work around it by taking React out of `strict` mode. Replace the code in your `index.js` file with the following code.
+
+```jsx
+import React from 'react';
+import ReactDOM from 'react-dom/client';
+import './index.css';
+import App from './App';
+import reportWebVitals from './reportWebVitals';
+
+const root = ReactDOM.createRoot(document.getElementById('root'));
+root.render(
+    <App />
+);
+
+// If you want to start measuring performance in your app, pass a function
+// to log results (for example: reportWebVitals(console.log))
+// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
```

Comment:
```suggestion
// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
```

this feels like a weird place for a bit.ly link. Pls disregard if you think I'm overreacting... Here's the actual link: https://create-react-app.dev/docs/measuring-performance/